### DISCUSSION TITLE
Re-built the Marlin 2.1.2.1 Firmware for a different from the original Artillery Genius Extruder-Thermistor

### DISCUSSION THREAD

**Discussion by fjherna** [dgtaheno/artillery-Genius#27] [General]
Hello,

I am using the **dgtaheno** Firmware Marlin 2.1.2.1 with great satisfaction.

However I have a new extruder (similar to Hemera) installed in my Artillery Genius. i.e. Trianglelab MATRIX EXTRUDER.

The current **dgtaheno** Firmware is working well. However I have a couple of things that I would like to modify into it.

1.- The THERMISTOR is different from the original and it needs a modification.
      Basically the modification is to change the "#define TEMP_SENSOR_0_1" to "#define TEMP_SENSOR_0_5"  at the "Configuration.h" file.

2.- The EXTRUDER is different because the direction of rotation of the engine is reversed.
      In order to fix it, the modification is to change the "#define INVERT_E0_DIR" value at "Configuration.h" for the following "#define INVERT_E0_DIR true".

After those changes I will need to re-build the Firmware again.

I have seen a link from **dgtaheno** that takes to the configurations directly from Marlin. i.e. "Configuration.h":
      https://github.com/MarlinFirmware/Configurations/tree/release-2.1.2.1/config/examples/Artillery
However I wonder if **dgtaheno** has added/modified any particular line at "Configuration.h" and/or "configuration_adv.h" in other to build his firmware.

Because I am a newbie with this kind of things, I will really appreciate if anyone could help me with it.

Kind regards,

fjherna1

**Comment by dgtaheno** [comment]
Hi,

I moved the discussion from issue to conversation, as it is not really a FW issue.

I have compiled Artillery Genius firmware 2.1.2.1 considering the 2 adjustments you mentioned, hope that it will work well for you.

[Custom Firmware.](https://we.tl/t-vyDBPcRZ72)

Please note that the download link will be only available for 7 days.

Best regards.

**Comment by fjherna** [comment]
Hi,

Thanks a lot. My fault. I did not realize that this group is for FW.

Thanks for the .hex file.

I guess that file is without BLT, isn't it?
:)

I am right now in the process of installing my brand new Antclabs BLTouch and my new BTT TFT35-E3 V3.0 Display Touch Screen Two Working Modes × 1 is now at the custom.
:)

I will report back to you when I will test your .hex file

Kind regards,

fjherna

**Reply by fjherna** [reply]
Hi,

I promised to report back after testing your .hex file.

I have tested it and it works. However it does not have BLTouch support that is what I have in my Artillery Genius now.

So I can use the Firmware without BLT support.
:)

Thanks and regards,

fjherna
